app/src/main/java/ayds/newyork/songinfo/moredetails/fulllogic/MoreDetailsWindow.kt

```diff
@@ -0,0 +1,137 @@
+private const val BASE_URL = "https://api.nytimes.com/svc/search/v2/"
+private const val RESPONSE = "response"
+private const val DOCS = "docs"
+private const val ABSTRACT = "abstract"
+private const val WEB_URL = "web_url"
+private const val NO_RESULTS = "No Results"
+private const val HTML_DIV_WIDTH = "<html><div width=400>"
+private const val HTML_FONT = "<font face=\"arial\">"
+private const val HTML_END_TAGS = "</font></div></html>"
+private const val IMAGE_URL =
+    "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRVioI832nuYIXqzySD8cOXRZEcdlAj3KfxA62UEC4FhrHVe0f7oZXp3_mSFG7nIcUKhg&usqp=CAU"
+
+class MoreDetailsWindow : AppCompatActivity() {
+    private lateinit var articlePane: TextView
+    private val artistInfoStorage: ArtistInfoStorage = ArtistInfoStorageImpl(this)
+
+    companion object {
+        const val ARTIST_NAME_EXTRA = "artistName"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_other_info)
+        articlePane = findViewById(R.id.textPane2)
+        getArtistInfo((intent.getStringExtra(ARTIST_NAME)))
+    }
+
+    private fun getArtistInfo(artistName: String?) {
+        Thread {
+            var artistInfo = artistName?.let { artistInfoStorage.getArtistInfo(it) }
+            artistInfo = if (artistInfo != null)
+                "[*]$artistInfo"
+            else
+                searchWithExternalService(artistName)
+            updateArtistData(artistInfo)
+        }.start()
+    }
+
+    private fun searchWithExternalService(artistName: String?): String {
+        lateinit var infoToReturn: String
+            try {
+                val response = getNYAPI().getResponse(artistName)
+                val abstract = response.getAbstract()
+                val url = response.getUrl()
+                infoToReturn = if (abstract == null)
+                    NO_RESULTS
+                else
+                    saveArtistInLocalStorage(artistName, abstract)
+                updateUrlButton(url)
+            } catch (e1: IOException) {
+                e1.printStackTrace()
+            }
+        return infoToReturn
+    }
+
+    private fun NYTimesAPI.getResponse(artistName: String?): JsonObject {
+        val callResponse = this.getArtistInfo(artistName).execute()
+        val jobj = Gson().fromJson(callResponse.body(), JsonObject::class.java)
+        return jobj[RESPONSE].asJsonObject
+    }
+
+    private fun JsonObject.getAbstract() = this[DOCS].asJsonArray[0].asJsonObject[ABSTRACT]
+
+    private fun JsonObject.getUrl() = this[DOCS].asJsonArray[0].asJsonObject[WEB_URL]
+
+    private fun saveArtistInLocalStorage(
+        artistName: String?,
+        abstract: JsonElement
+    ): String {
+        var info = abstract.asString.replace("\\n", "\n")
+        info = textToHtml(info, artistName)
+        artistInfoStorage.saveArtist(artistName, info)
+        return info
```

Comment:
Siempre que se pueda, evitar las var. 
```suggestion
        val cleanAbstract = abstract.asString.replace("\\n", "\n")
        val info = textToHtml(cleanAbstract,  artistName)
        artistInfoStorage.saveArtist(artistName, info)
        return info
```